File: backend/ngts/src/main/java/com/three/ngts/Entity/Seat.java
package com.three.ngts.Entity;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "seat")
public class Seat {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "seat_id")
    private Long seatId;

    @Column(name = "seat_no")
    private String seatNo;

    @ManyToOne
    @JoinColumn(name = "schedule_id")
    private BusSchedule busSchedule;

    @ManyToOne
    @JoinColumn(name = "ticket_id")
    private Ticket ticket;
}

File: frontend/next-generation-ticketing-system/src/Entity/Seat.ts
import BusSchedule from "./BusSchedule";
import Ticket from "./Ticket";

class Seat {
  seatId: number;
  seatNo: string;
  busSchedule: BusSchedule;
  ticket: Ticket;

  constructor(
    seatId: number,
    seatNo: string,
    busSchedule: BusSchedule,
    ticket: Ticket
  ) {
    this.seatId = seatId;
    this.seatNo = seatNo;
    this.busSchedule = busSchedule;
    this.ticket = ticket;
  }
}

export default Seat;
